hedera-mirror-rest/service/transactionResultService.js

```diff
@@ -0,0 +1,98 @@
+/*-
+ * ‌
+ * Hedera Mirror Node
+ * ​
+ * Copyright (C) 2019 - 2021 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+'use strict';
+
+const _ = require('lodash');
+
+const {DbError} = require('../errors/dbError');
+const {InvalidArgumentError} = require('../errors/invalidArgumentError');
+const {TransactionResult} = require('../model');
+
+/**
+ * Transaction results retrieval business logic
+ */
+class TransactionResultService {
+  constructor() {
+    this.transactionResultToProtoMap = new Map();
+    this.transactionResultProtoToResultMap = new Map();
+  }
+
+  static transactionResultsQuery = `select ${TransactionResult.PROTO_ID}, ${TransactionResult.RESULT}
+                                    from ${TransactionResult.tableName}`;
+
+  populateTransactionResultMaps(transactionResults) {
+    this.transactionResultToProtoMap = new Map();
+    this.transactionResultProtoToResultMap = new Map();
+
+    transactionResults.forEach((transactionResult) => {
+      this.transactionResultToProtoMap.set(transactionResult.result, transactionResult);
+      this.transactionResultProtoToResultMap.set(transactionResult.protoId, transactionResult);
+    });
+  }
+
+  async getTransactionResults() {
+    let rows;
+    try {
+      const result = await pool.query(TransactionResultService.transactionResultsQuery);
+      rows = result.rows;
+    } catch (err) {
+      this.promise = null;
+      throw new DbError(err.message);
+    }
```

Comment:
`this.promise` is never used.

since after the refactor, load becomes a one-shot effort at startup, we should just fail without retry so imo the `promise` is no longer needed, then we can simplify it to:

```suggestion
    const {rows} = await utils.queryQuitely(TransactionResultService.transactionResultsQuery);
```

Reply:
Yeah that was the intention, forgot to simplify. Will update, thanks

Reply:
Actually, there's a circular dependency which is why I don't use the `utils.queryQuietly`.
Will remove the promise though

Reply:
I see. can leave circular dependency issue to a different ticket